plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java

```diff
@@ -86,8 +86,14 @@ public DBSDataType getLocalDataType(String typeName) {
             affinity = SQLiteAffinity.BLOB;
         } else if (typeName.startsWith("REAL") || typeName.startsWith("FLOA") || typeName.startsWith("DOUB")) {
             affinity = SQLiteAffinity.REAL;
-        } else {
+        } else if (typeName.toUpperCase().contains(SQLConstants.DATA_TYPE_INT) || typeName.toUpperCase().contains("NUMERIC") ||
```

Comment:
Maybe cache the result of `typeName.toUpperCase()`?

Reply:
it's already in upper case, actually, so no needed here
fixed